cfgrammar/src/lib/span.rs

```diff
@@ -44,72 +44,98 @@ impl Span {
     }
 }
 
-pub struct NewlineToLineColCache<'a> {
-    src: &'a str,
-    high_water: usize,
+pub struct NewlineToLineColCache {
     newlines: Vec<usize>,
+    trailing_bytes: usize,
```

Comment:
That's very clever. I hadn't even thought of this!

Reply:
Hopefully not too clever, there may be some more places we need to use it beyond where we currently do, I'll try and think about this tonight, and come up with some tests which deal with multiple strings, including needing the trailing_bytes.